crates/iceberg/src/spec/values.rs

```diff
@@ -3439,11 +3443,13 @@ mod tests {
                     "bar".to_string(),
                 ))),
                 None,
+                Some(Literal::Primitive(PrimitiveLiteral::Int(1000))),
             ])),
             &Type::Struct(StructType::new(vec![
                 NestedField::required(2, "id", Type::Primitive(PrimitiveType::Int)).into(),
                 NestedField::optional(3, "name", Type::Primitive(PrimitiveType::String)).into(),
                 NestedField::optional(4, "address", Type::Primitive(PrimitiveType::String)).into(),
+                NestedField::required(5, "extra", Type::Primitive(PrimitiveType::Int)).into(),
```

Comment:
Hi, would you like to add a new test that cover the mis-order cases?

Reply:
Yes, but I find that it can also pass originally. I'm trying to find the test case that can't pass originally.

Reply:
> Yes, but I find that it can also pass originally. I'm trying to find the test case that can't pass originally.

Thank you, that will be really meaningful.

Reply:
I have found the reason why this can pass originally: the avro writer will ensure the record order according to the schema: https://github.com/apache/avro-rs/blob/390a150bfc5999eb852c9c0ef40335612f1407b5/avro/src/encode.rs#L247.

However, if we serialize into other formats, e.g. json, the order can't be guaranteed.